nav2_util/include/nav2_util/get_robot_pose_client.hpp

```diff
@@ -29,8 +31,17 @@ class GetRobotPoseClient : public ServiceClient<nav2_msgs::srv::GetRobotPose>
   {
   }
 
-  explicit GetRobotPoseClient(rclcpp::Node::SharedPtr & node)
-  : ServiceClient<nav2_msgs::srv::GetRobotPose>("GetRobotPose", node)
+  explicit GetRobotPoseClient(
+    rclcpp::Node::SharedPtr node,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr)
+  : ServiceClient<nav2_msgs::srv::GetRobotPose>("GetRobotPose", node, group)
+  {
+  }
+
+  explicit GetRobotPoseClient(
+    rclcpp_lifecycle::LifecycleNode::SharedPtr node,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr)
+  : ServiceClient<nav2_msgs::srv::GetRobotPose>("GetRobotPose", node, group)
   {
```

Comment:
I noticed that there is not a constructor for the node_interfaces version. With node helper classes like this, we should standardize on the constructors we provide. ROS2 seems to provide the rclcpp::Node version and the node_interfaces version. I think there are three options:

1. All Three - rclcpp::Node, rclcpp_lifecycle::LifecycleNode, and node_interfaces
2. Node and interfaces (like ROS2)
3. Node and LifecycleNode

Let's talk as a group to decide.